operator/apis/operator/v1alpha1/banyandb_types.go

```diff
@@ -0,0 +1,66 @@
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type BanyanDBSpec struct {
+
+	// Version of BanyanDB.
+	// +kubebuilder:validation:Required
+	Version string `json:"version"`
+
+	// Image is the BanyanDB Docker image to deploy.
+	// +kubebuilder:validation:Required
+	Image string `json:"image,omitempty"`
+
+	// BanyanDB startup parameters
+	// +kubebuilder:validation:Optional
+	Config map[string]string `json:"config"`
+
+	// BanyanDB Service
+	// +kubebuilder:validation:Optional
+	Service Service `json:"service,omitempty"`
```

Comment:
It should be two different items: HTTP and gRPC. Please add a subnode to config TLS relevant parameters.

Reply:
Got it, I'll add it in the future.